lib/engine.js

```diff
@@ -24,6 +26,18 @@ module.exports.start = function start(configuration) {
   });
   configuration.masterID = masterID;
   configuration.storage = new Storage(configuration);
+  configuration.emitter = new Emitter();
+  // attach user event listeners
+  if (configuration.config.get('output:listeners')) {
```

Comment:
Worth considering if this should be nested under `output` (or if we need `output` namespace at all) and not just `listeners` or something. Potentially you could do anything you wanted on the CI on `master:end` for example (i.e. not just report/output type stuff).

Reply:
if there is going to be a refactor of the configuration spec, this is the time.

Reply:
To clarify my reasoning on using `output.listeners`: Events emit data from the process, so I thought of that as "output". Also, want to avoid cluttering the top level of the JSON config. So far it is:
* driver
* plugins
* data
* profiles
* output